### DISCUSSION TITLE
how to use env.js file in next.config.js file to read the variables

### DISCUSSION THREAD

**Discussion by kaleshasheik** [vercel/next.js#11252] [Help]
Hi ,

 I have created a file env.js with variable eg: API_URL = "https://apiurl.com"

In next.config.js, I have used as

require('dotenv').config()
const path = require('path')
const Dotenv = require('dotenv-webpack')

 webpack(config) {
      config.plugins = [
        ...config.plugins,
        new Dotenv({
          path: path.join(__dirname, 'env.js')

        })
      ]
    return config;
  }
});
After that, 
when i do console, its not printing the variables defined in env.js
console.log(process.env)

Can someone help on this ?

**Comment by timneutkens** [comment]
There's an open RFC to add `.env` support out of the box: https://github.com/zeit/next.js/discussions/11106
